stumpy/cache.py

```diff
@@ -190,32 +205,35 @@ def _recompile():
     return
 
 
-def _save():
+def _save(cache_dir):
     """
     Save all njit functions
 
     Parameters
     ----------
-    None
+    cache_dir : str
+        The path to the numba cache directory
 
     Returns
     -------
     None
     """
     _enable()
+    _clear(cache_dir)
```

Comment:
I was wondering if we should remove `_clear` from here. Not because it is a bad idea to clear before saving again but because it results in adding param `cache_dir` to the function `_save`, and it might become misleading. A user may think it can actually save to that directory. The function `cache.save` does have warning but what if user does not even change the default value of `numba.config.CACHE_DIR`, and just pass a path to `cache.save`?

Reply:
> it might become misleading. A user may think it can actually save to that directory. 

I think that is a fair point. I accept. This means that we must remind the user to always `cache.clear()` first